modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1463,14 +1466,141 @@ export class Wallet implements IWallet {
+   * @param {string} [params.shareOptions[].keychain.path] - The derivation path of the keychain.
+   * @param {string} params.shareOptions[].user - The user to share the wallet with.
+   * @param {string} params.shareOptions[].permissions - The permissions granted to the user.
+   * @param {string} [params.walletPassphrase] - The wallet passphrase used to decrypt the keychain.
+   * @throws {Error} If `shareOptions` is empty, or if required keychain parameters (`toPubKey` and `path`) are missing when needed.
+   * @throws {Error} If unable to decrypt the user keychain.
+   * @returns {Promise<CreateBulkWalletShareListResponse>} A promise that resolves with the response of the bulk wallet share creation.
+   */
+  async bulkWalletShare(params: BulkWalletShareOptions): Promise<CreateBulkWalletShareListResponse> {
+    if (!params.shareOptions || Object.keys(params.shareOptions).length === 0) {
+      throw new Error('shareOptions cannot be empty');
+    }
+    const bulkCreateShareOptions: BulkCreateShareOptions = { shareOptions: [] };
+
+    for (let index = 0; index < params.shareOptions.length; index++) {
+      const shareOption = params.shareOptions[index];
+
+      common.validateParams(shareOption, ['userId', 'pubKey', 'path'], []);
+
+      const needsKeychain = shareOption.permissions && shareOption.permissions.indexOf('spend') !== -1; // Remove !shareOption.keychain
+      let sharedKeychain: any = undefined;
+
+      if (needsKeychain) {
+        try {
+          const keychain = await this.getEncryptedUserKeychain();
+
+          if (!keychain.encryptedPrv) {
+            sharedKeychain = undefined;
+          } else {
+            const userPrv = decryptKeychainPrivateKey(this.bitgo, keychain, params.walletPassphrase);
+            if (!userPrv) {
+              throw new Error('Unable to decrypt user keychain.');
+            }
+
+            const eckey = makeRandomKey();
+            const sharing = shareOption;
+            assert(sharing.pubKey);
+
+            const secret = getSharedSecret(eckey, Buffer.from(sharing.pubKey, 'hex')).toString('hex');
+            const newEncryptedPrv = this.bitgo.encrypt({ password: secret, input: userPrv });
+
+            let pub = keychain.pub ?? keychain.commonPub;
+            if (keychain.commonKeychain) {
+              pub =
+                this.baseCoin.getMPCAlgorithm() === 'eddsa'
+                  ? EddsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain)
+                  : EcdsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain);
+            }
+
+            sharedKeychain = {
+              pub,
+              encryptedPrv: newEncryptedPrv,
+              fromPubKey: eckey.publicKey.toString('hex'),
+              toPubKey: sharing.pubKey,
+              path: sharing.path,
+            };
+          }
+        } catch (e) {
+          if (e.message === 'No encrypted keychains on this wallet.') {
+            sharedKeychain = {};
```

Comment:
why are you not re-throwing this error?

Reply:
This is the reason - https://github.com/BitGo/BitGoJS/blob/master/modules/sdk-core/src/bitgo/wallet/wallet.ts#L1527-L1531

Reply:
not a fan of handling errors like this, but no alternative comes to mind at the moment.